pkg/plugins/golang/v3/scaffolds/internal/templates/dockerfile.go

```diff
@@ -55,11 +58,12 @@ COPY api/ api/
 COPY controllers/ controllers/
 
 # Build
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o manager main.go
+# When TARGETOS and TARGETARCH are not informed then, the default values used are linux and amd64
+RUN CGO_ENABLED=0 GOOS=${TARGETOS:-linux} GOARCH=${TARGETARCH:-amd64} go build -a -o manager main.go
```

Comment:
HI @aleskandro,

The `--platform` flag does not require BuildKit. You only need BuildKit when you inform a list of flags.
See that the code here is the same as before. 

We are only allowing the author informs the GOOS and GOARCH if they wish to say for the Golang used it to build the image. It is instead of using Linux and amd64 fixed as before we know to allow the user customize it if/when they need. 

Current code: `RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o manager main.go`
Then, this PR: `RUN CGO_ENABLED=0 GOOS=${TARGETOS:-linux} GOARCH=${TARGETARCH:-amd64} go build -a -o manager main.go`

So, I am not sure what issue this can cause. Because if someone customize their target to use platform, TARGETOS and TARGETARCH that do not match then it is up to them. We are only providing the option to help out.

Reply:
I think we spoke about that. 
Currently the same values are fixed we just need to set the args to allow the buildx work